.github/workflows/miniconda_templated.yaml

```diff
@@ -2,69 +2,90 @@ name: Build and Push miniconda templated containers
 
 on:
   push:
+    branches:
+      - master
+  pull_request:
     paths:
-      - "atlas-schemas-base/*"
+      - "*/env.yaml"
+      - "*/image_tag"
+      - "*/post_install_dockerfile"
+      - "miniconda_template/*"
+      - ".github/workflows/miniconda_template.yaml"
 
 jobs:
-  test:
-    name: Build image
+  getchanges:
+    name: Get changed repos
     runs-on: ubuntu-20.04
-
-    if: startsWith( github.ref, 'refs/heads/feature/' )
     steps:
-    - uses: actions/checkout@v2
-
-    - name: Read image
-      id: gettag
-      run: echo "::set-output name=image::$(head -n 1 atlas-schemas-base/image_tag | awk -F':' '{ print $1 }' )"
-
-    - name: Read tag
-      id: gettagver
-      run: echo "::set-output name=tag::$(head -n 1 atlas-schemas-base/image_tag | awk -F':' '{ print $2 }' )"
-
-    - name: Assemble Dockerfile
-      run: miniconda_template/create_dockerfile.sh atlas-schemas-base
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v11.5
+        with:
+          separator: ','
+      - name: Changed base directories for u-mamba based containers
+        id: changes
+        run: |
+          CHANGED_DIRS_JSON=$(echo ${{ steps.changed-files.outputs.all_changed_and_modified_files }} | \
+            tr ',' '\n' | \
+            awk '{ if ( $0 ~ /\// ) { print $1 } }' | \
+            awk -F'/' '{ print $1 }' | sort -u | \
+            awk 'system("test -f "$1"/env.yaml")==0 { print $1 }' | \
```

Comment:
So, if I'm reading right, u-mamba based images are identified only by the presence of an env.yaml? Maybe we could make the convention more specific, and require that env files be called something like 'micromamba-env.yaml' or similar?

Reply:
Yes, this is probably good to do given that CI depends on it.